Content.Client/Roles/RoleCodewordSystem.cs

```diff
@@ -0,0 +1,20 @@
+using Content.Shared.Roles;
+
+namespace Content.Client.Roles;
+public sealed class RoleCodewordSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeNetworkEvent<RoleCodewordEvent>(SetRoleCodewords);
+    }
+
+    private void SetRoleCodewords(RoleCodewordEvent args)
+    {
+        EntityUid mindId = GetEntity(args.Entity);
+
+        var comp = EnsureComp<RoleCodewordComponent>(mindId);
+        comp.RoleCodewords[args.RoleKey] = new CodewordsData(args.Color, args.Codewords);
+    }
```

Comment:
This should probably be done with component states/auto networking.

Reply:
Also, wouldn't the client be able to simply try to grab codewords from every player and find who is a traitor?

Reply:
The component is only added to the mind entity of the traitor player, on the traitor player's client via a server-initiated directed network event. Other clients don't have this network event raised and because of that don't even get the codewords sent to them.